src/dymaptic.GeoBlazor.Core/Components/MapComponent.razor.cs

```diff
@@ -240,6 +271,421 @@ protected virtual async Task RenderView(bool forceRender = false)
+    /// </remarks>
+    public Task AddReactiveWatcher<T>(string watchExpression, Func<T, Task> handler, string? targetName = null,
+        bool once = false, bool initial = false)
+    {
+        return AddReactiveWatcherImplementation(watchExpression, handler, targetName, once, initial);
+    }
+    
+    /// <summary>
+    ///     Tracks any properties accessed in the <code>watchExpression</code> and calls the callback when any of them change.
+    /// </summary>
+    /// <param name="watchExpression">
+    ///     Expression used to get the current value. All accessed properties will be tracked.
+    /// </param>
+    /// <param name="handler">
+    ///     The function to call when there are changes.
+    /// </param>
+    /// <param name="targetName">
+    ///     The name of the target you are referencing in the <code>watchExpression</code>. For example, if the expression is "layer?.refresh", then the targetName should be "layer". The type of the target should also match the class on which this method was called.
+    /// </param>
+    /// <param name="once">
+    ///     Whether to fire the callback only once.
+    /// </param>
+    /// <param name="initial">
+    ///     Whether to fire the callback immediately after initialization, if the necessary conditions are met.
+    /// </param>
+    /// <typeparam name="T">
+    ///     The type of return value to expect in the handler.
+    /// </typeparam>
+    /// <exception cref="UnMatchedTargetNameException">
+    ///     This exception is thrown when a watchExpression's target object name doesn't match the targetName parameter.
+    /// </exception>
+    /// <remarks>
+    ///     For adding watchers to types other than <see cref="MapView"/> and <see cref="SceneView"/>, the default <code>targetName</code> should not be relied upon. Make sure it matches the variable in your <code>watchExpression</code>
+    /// </remarks>
+    // ReSharper disable once UnusedMethodReturnValue.Global
+    public Task AddReactiveWatcher<T>(string watchExpression, Action<T> handler, string? targetName = null, 
+        bool once = false, bool initial = false)
+    {
+        return AddReactiveWatcherImplementation(watchExpression, handler, targetName, once, initial);
+    }
+
+    private async Task AddReactiveWatcherImplementation(string watchExpression, Delegate handler, string? targetName, 
+        bool once, bool initial)
+    {
+        string typeName = GetType().Name;
+        targetName ??= typeName.Contains("View") ? "view" : typeName.ToLowerFirstChar();
+
+        if (!watchExpression.Contains(targetName))
+        {
+            throw new UnMatchedTargetNameException(targetName, watchExpression);
+        }
+        IJSObjectReference? jsRef = 
+            await JsModule!.InvokeAsync<IJSObjectReference?>("addReactiveWatcher", Id, 
+                targetName, watchExpression, once, initial);
+
+        if (jsRef != null)
+        {
+            _watchers[watchExpression] = (handler, jsRef);
```

Comment:
This is limiting to one handler per watchExpression?

Reply:
That's correct. I can't imagine wanting to hook up multiple, when they would fire simultaneously, as opposed to having a single handler that did multiple things.